zookeeper-server/src/main/java/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -2268,7 +1885,7 @@ public Stat exists(final String path, Watcher watcher) throws KeeperException, I
      * @throws InterruptedException If the server transaction is interrupted.
      */
     public Stat exists(String path, boolean watch) throws KeeperException, InterruptedException {
-        return exists(path, watch ? watchManager.defaultWatcher : null);
+        return exists(path, watch ? defaultWatcher : null);
     }
```

Comment:
> let ZooKeeper syncs a copy of default watcher, in order to reduce dependencies to getWatchManager()

 You have used `getWatchManager()` in the `zookeeper.java`, `getWatchManager().getDefaultWatcher()` is better? The origin logic is: get the `defaultWatcher` from `ZKWatchManager`.

Reply:
Yes definitely. Updating...